src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -132,34 +146,57 @@
      * The client secret from the OpenShift server.
      */
     private final Secret clientSecret;
+    
+    /**
+     * This map contains defaults gathered for this objects fields when those fields were not set in the UI.
+     * We don't store our inferred defaults in those fields, because if we do, they will show up in the gui
+     * When the getters are called.
+     */
+    private final Map<String,String> overrides = new HashMap<String,String>();
+    
 
     @DataBoundConstructor
     public OpenShiftOAuth2SecurityRealm(String serviceAccountDirectory, String serviceAccountName, String serverPrefix, String clientId, String clientSecret) throws IOException, GeneralSecurityException {
         HttpTransport transport = HTTP_TRANSPORT;
+        
+        LOGGER.fine(String.format("ctor args sa dir %s sa name %s svr prefix %s client id %s client secret %s", serviceAccountDirectory, serviceAccountName, serverPrefix, clientId, clientSecret));
 
-        this.serviceAccountDirectory = Util.fixEmpty(serviceAccountDirectory);
-    	if (Util.fixEmpty(serviceAccountName) == null)
-    		this.serviceAccountName = "jenkins";
-    	else
-    		this.serviceAccountName = serviceAccountName;
-    	// TODO: should this be in a different method?  Will this show up in settings page?
-    	if (this.serviceAccountDirectory != null) {
-    		if (Util.fixEmpty(clientSecret) == null) {
-	            BufferedReader r = new BufferedReader(new FileReader(new File(this.serviceAccountDirectory, "token")));
-	            clientSecret = r.readLine();
+        String fixedServiceAccountDirectory = Util.fixEmpty(serviceAccountDirectory);
+        String fixedClientId = Util.fixEmpty(clientId);
+        String fixedClientSecret = Util.fixEmpty(clientSecret);
+        String fixedServerPrefix = Util.fixEmpty(serverPrefix);
+        
+        // decided not to default this since it is not even consistent within OpenShift;
+        // going through the example instantapp templates, the SA will be "default";
+        // going through the auto generated server, the SA will be "jenkins"
+        this.serviceAccountName = Util.fixEmpty(serviceAccountName);
```

Comment:
oc get user ~ does not work because the shell expands ~ to $HOME.

On Wed, Aug 10, 2016 at 1:23 PM, Gabe Montero <notifications@github.com>
wrote:

> In src/main/java/org/openshift/jenkins/plugins/openshiftlogin/
> OpenShiftOAuth2SecurityRealm.java
> <https://github.com/openshift/jenkins-openshift-login-plugin/pull/2#discussion_r74289644>
> :
>
> > -    	else
> > -    		this.serviceAccountName = serviceAccountName;
> > -    	// TODO: should this be in a different method?  Will this show up in settings page?
> > -    	if (this.serviceAccountDirectory != null) {
> > -    		if (Util.fixEmpty(clientSecret) == null) {
> > -	            BufferedReader r = new BufferedReader(new FileReader(new File(this.serviceAccountDirectory, "token")));
> > -	            clientSecret = r.readLine();
> > +        String fixedServiceAccountDirectory = Util.fixEmpty(serviceAccountDirectory);
> > +        String fixedClientId = Util.fixEmpty(clientId);
> > +        String fixedClientSecret = Util.fixEmpty(clientSecret);
> > +        String fixedServerPrefix = Util.fixEmpty(serverPrefix);
> > +
> > +        // decided not to default this since it is not even consistent within OpenShift;
> > +        // going through the example instantapp templates, the SA will be "default";
> > +        // going through the auto generated server, the SA will be "jenkins"
> > +        this.serviceAccountName = Util.fixEmpty(serviceAccountName);
>
> OK, oc get user ~ did not work, but oc get user/~ does. That said, the
> error I was getting when trying in Java was not the same as the command
> line. I still get permission denied in Java. But using the --loglevel=8
> output, I should be able to work backwards and get something to work. Might
> have to ditch the openshift-restclient-java in the process and go with
> lower level okhttp packages, but no biggie.
>
> Thanks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/jenkins-openshift-login-plugin/pull/2/files/2c7cc23ad8dea53bbdfaf093ea54ea5d7d854700#r74289644>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p0jgI1KZH535B272heqvuS94D6Jeks5qegkugaJpZM4Jfdh->
> .
>